Changelog for RabbitMQ 3.8 update

rouazana opened the pull request:
Related to https://github.com/linagora/james-project/pull/2776

Arsnael commented on upgrade-instructions.md:
```diff
@@ -16,17 +16,28 @@ Changes to apply between 3.4.x and 3.5.x will be reported here.
+
+The distributed James project (relying on Guice, Cassandra, ElasticSearch, RabbitMQ and optionally Swift) benefits from a new distributed task mananger.
+
+For the plain operation if this new task manager, James now relies at least on RabbitMQ 3.8. So please upgrade your RabbitMQ version if you use this component.
```
s/if/of ?

chibenwa replied:
We should not speak about components here but about products. How as a user do I know which component I run?

Arsnael replied:
Then reformulate that cause grammatically speaking, it's wrong

Arsnael replied:
if `using` this new task manager?

rouazana commented:
:+1:

remk commented on upgrade-instructions.md:
```diff
@@ -16,17 +16,28 @@ Changes to apply between 3.4.x and 3.5.x will be reported here.
+
+The distributed James project (relying on Guice, Cassandra, ElasticSearch, RabbitMQ and optionally Swift) benefits from a new distributed task mananger.
+
+For the plain operation if this new task manager, James now relies at least on RabbitMQ 3.8. So please upgrade your RabbitMQ version if you use this component.
```
s/if/of ?

remk commented on upgrade-instructions.md:
```diff
@@ -29,9 +29,13 @@ Date 26/11/2019
 The distributed James project (relying on Guice, Cassandra, ElasticSearch, RabbitMQ and optionally Swift) benefits from a new distributed task mananger.
 
-For the plain operation if this new task manager, James now relies at least on RabbitMQ 3.8. So please upgrade your RabbitMQ version if you use this component.
+In order to enforce task sequential processing, we rely on an exclusive consumer, which is a feature introduced in RabbitMQ 3.8.
```
s/an exclusive consumer/a single active consumer

chibenwa replied:
Naming the rabbitMQ 3.8 feature explicitly is intendeed @remk .

Maybe we can say : `In order to enforce task sequential processing via a signle active consumer at the cluster level, we rely on an exclusive consumer, which is a feature introduced in RabbitMQ 3.8`

chibenwa commented on upgrade-instructions.md:
```diff
@@ -33,7 +33,7 @@ Concerned products: Guice distributed James (rabbitMQ)
 The distributed James project (relying on Guice, Cassandra, ElasticSearch, RabbitMQ and optionally Swift) benefits from a new distributed task mananger.
 
-In order to enforce task sequential processing, we rely on an exclusive consumer, which is a feature introduced in RabbitMQ 3.8.
+In order to enforce task sequential processing, we rely on a single active consumer, which is a feature introduced in RabbitMQ 3.8.
```
`In order to enforce task sequential processing via a signle active consumer at the cluster level, we rely on an exclusive consumer, which is a feature introduced in RabbitMQ 3.8`

rouazana replied:
ok I'm doing the update (and a squash because the fixup are not referencing the right commit)

rouazana replied:
in fact not, the naming is the good one: http://next.rabbitmq.com/consumers.html#single-active-consumer
but I will add `at the cluster level` (and do the squash)

chibenwa commented on upgrade-instructions.md:
```diff
@@ -16,17 +16,32 @@ Changes to apply between 3.4.x and 3.5.x will be reported here.
+
+In order to enforce task sequential processing at the cluster level, we rely on a single active consumer, which is a feature introduced in RabbitMQ 3.8.
+
+Users of distributed James product thus need to upgrade their RabbitMQ server to be at least version 3.8.
```
Please explain that:
 - The migration require the task manager work queue manual deletion
 - That in order to avoid data loss, migration should be done when no task is in progress or awaiting.

rouazana replied:
in fact this is not needed because we should not document changes between 2 releases, no?

Arsnael commented on upgrade-instructions.md:
```diff
@@ -37,6 +37,10 @@ In order to enforce task sequential processing at the cluster level, we rely on
 
 Users of distributed James product thus need to upgrade their RabbitMQ server to be at least version 3.8.
 
+To perform the upgrade the RabbitMQ queue `taskManagerWorkQueue` have to be deleted. James will then recreate it with the single active consumer property.
```
s/have/has

Arsnael commented on upgrade-instructions.md:
```diff
@@ -37,6 +37,10 @@ In order to enforce task sequential processing at the cluster level, we rely on
 Users of distributed James product thus need to upgrade their RabbitMQ server to be at least version 3.8.
 
+To perform the upgrade the RabbitMQ queue `taskManagerWorkQueue` have to be deleted. James will then recreate it with the single active consumer property.
+In order not to lose any messages, all the messages should have been processed, thus this queue should be empty.
```
s/not to lose/to not loose?

remk replied:
No, here we want the verb : `to lose / lost / lost`
not the adjective `Loose` meaning untight

Arsnael replied:
ah yeah, my bad :)

mbaechler commented on upgrade-instructions.md:
```diff
@@ -16,17 +16,36 @@ Changes to apply between 3.4.x and 3.5.x will be reported here.
  - [ElasticSearch performance enhancements](#elasticsearch-performance-enhancements)
  - [JAMES-2703 Post 3.4.0 release removals](#james-2703-post-340-release-removals)
 
+#### RabbitMQ minimal version
```
Did we wrote an ADR for that?

rouazana replied:
that what? minimal version or new workqueue design?

mbaechler replied:
the design decision that leads to requiring Rabbit 3.8 (the usage of single active consumer)

chibenwa commented on upgrade-instructions.md:
```diff
@@ -16,17 +16,36 @@ Changes to apply between 3.4.x and 3.5.x will be reported here.
+
+Users of distributed James product thus need to upgrade their RabbitMQ server to be at least version 3.8.
+
+To perform the upgrade the RabbitMQ queue `taskManagerWorkQueue` has to be deleted. James will then recreate it with the single active consumer property.
```
`To perform the upgrade the RabbitMQ queue *taskManagerWorkQueue* has to be deleted.`

How?

rouazana replied:
I will remove this fixup, it's useless as it does not concern a released modification.
(for the record, the answer of this question is: using rabbitmq web management interface)

rouazana commented:
I removed the last fixups (about upgrade procedure which should not be documented) and squashed.
The build can already be considered as green.

Concerning the missing adr, I would do it in an other PR (superseding adr 0003).
